### DISCUSSION TITLE
Add device address to "device card"

### DISCUSSION THREAD

**Discussion by danielperna84** [SukramJ/aiohomematic#859] [Ideas]
**Is your feature request related to a problem? Please describe.**
When looking at the device an entity belongs to, I'd also like to see the devices address (`MEQ...`, `001122...`), not just the type and firmware. Now that names are used I'd first have to look into the CCU to get the real address.

**Describe the solution you'd like**
Add the device address to the card where we also see the firmware.

**Additional context**
Screenshot of where I would expect the address.:
<img width="254" alt="gi" src="https://user-images.githubusercontent.com/7396998/149577296-f439b49d-38a5-4896-9ada-b00c29c63b63.png">

**Comment by SukramJ** [comment]
I don't think this is possible. [See](https://developers.home-assistant.io/docs/device_registry_index#device-properties) 

You can see the channel address at every entities attributes.

**Comment by danielperna84** [comment]
> identifiers
> Set of (DOMAIN, identifier) tuples. Identifiers identify the device in the outside world. An example is a serial number.

Wouldn't this be the correct property? I guess it then would be filled with `('hahm', '12345')`. I don't know if that would be displayed in the UI though. 🤔

And yes, of course it can be seen in the attributes. But that seems less intuitive to me. I can see other users asking where they'll find the address.

Edit: for my Hue bulb connected via zigbee2mqtt it looks like this:
<img width="242" alt="huebulb" src="https://user-images.githubusercontent.com/7396998/149579193-c40b7503-e69b-41ed-b84b-756b4a667945.png">

**Comment by SukramJ** [comment]
This property is not displayed.


This is what is used :

    @property
    def device_info(self) -> dict[str, Any]:
        """Return device specific attributes."""
        return {
            "identifiers": {
                (
                    self._central.domain,
                    f"{self._device_address}{IDENTIFIERS_SEPARATOR}{self._interface_id}",
                )
            },
            "name": self.name,
            "manufacturer": MANUFACTURER,
            "model": self.device_type,
            "sw_version": self.firmware,
            "suggested_area": self.room,
            "via_device": (self._central.domain, self._central.instance_name),
        }

**Comment by danielperna84** [comment]
So I guess what I'm seeing with my bulbs is specific to the MQTT integration, not a standard. Ok, then we probably should not put the address there.

**Comment by SukramJ** [comment]
Any further ideas to display the SN?
What we **not** should do is to abuse existing fields like model.

**Reply by noxhirsch** [reply]
I just found the "Hardware" section on a Shelly device that I hadn't noticed before. Wouldn't that be a suitable place for the SN?
![image](https://github.com/danielperna84/hahomematic/assets/30938717/7f4935c8-1393-4b1f-a30d-e8b49f10cd95)

**Reply by SukramJ** [reply]
The UI element `Hardware` is named `hw_version` in the model.
So using `hw_version` to store a serial number is not correct.

**Comment by danielperna84** [comment]
I think as long as HA does not provide an appropriate field for this, we should not hack around this. My suggestion is to label this as _wontfix_ and keep it open. This allows others to see we are aware of this, and why it is not implemented yet. At least I think from time to time someone else could have the same idea and ask for this. And maybe in the future HA adds a dedicated field for this, and then it can be implemented.

**Comment by ThomDietrich** [comment]
I agree, this would be nice to have.

Did you look at further integrations? ZHA for example:

![image](https://user-images.githubusercontent.com/2870104/162332886-3323dc7e-9667-48e6-b9a4-a18b2246cc16.png)

**Comment by SukramJ** [comment]
No, i don't. This is a **custom UI** for ZHA and not the default one.

**Comment by SukramJ** [comment]
Die "device card" erhält mit HA 2023.11 ein neues Feld `serial number`. Da wird dann bei Geräten die `device address` bzw. bei der CCU die Seriennummer des RF-Moduls drin sein.

**Reply by noxhirsch** [reply]
Top - vielen Dank! :)

**Reply by ThomDietrich** [reply]
Nice! Danke ;)
